File: leetcode/01678_goal-parser-interpretation/1678-goal-parser-interpretation.cpp
class Solution {
    public:
        string interpret(string command) {
            string newCommand= "";
            for(int i=0; i<command.size(); i++)
            {
                if (command[i] == 'G') {
                    newCommand += command[i];
                } else if (command[i] == '(' && command[i + 1] == ')') {
                    newCommand += "o";
                    i++;
                } else if (command[i] == '(' && command[i + 1] == 'a') {
                    newCommand += "al";
                    i += 3;
                }
            }
            return newCommand;
        }
    };

File: leetcode/01678_goal-parser-interpretation/1678-goal-parser-interpretation.py
class Solution:
    def interpret(self, command: str) -> str:
        newCommand = ""
        for i in range(len(command)):
            if command[i] == "G":
                newCommand += command[i]
            elif command[i] == "(" and command[i + 1] == ")":
                newCommand += "o"
                i += 1
            elif command[i] == "(" and command[i + 1] == "a":
                newCommand += "al"
                i += 3

        return newCommand
